### PR TITLE
Lb tests

### PR DESCRIPTION
**Pull request by KaiSzuttor**
Description of changes:
 - minor code improvements for LB
 - added streaming test for LB

### CODE REVIEW

**Comment by KaiSzuttor** [comment]
osx vms do currently not work @mkuron

**Comment by mkuron** [comment]
@KaiSzuttor, OS X VMs fixed this morning: https://github.com/espressomd/espresso/pull/2132#issuecomment-409852827. Your current broken Docker builds are fixed by https://github.com/espressomd/espresso/pull/2134.

File: maintainer/gh_post_status.sh [hunk 1]
```diff
@@ -8,4 +8,4 @@ STATUS="$1"
 curl "https://api.github.com/repos/espressomd/espresso/statuses/$GIT_COMMIT?access_token=$GITHUB_TOKEN" \
      -H "Content-Type: application/json" \
      -X POST \
-     -d "{\"state\": \"$STATUS\", \"context\": \"ICP GitLab CI\", \"target_url\": \"$URL\"}"
+     -d "{\"state\": \"$STATUS\", \"context\": \"ICP GitLab CI\", \"target_url\": \"$URL\", \"description\": \"GitLab CI is performing tests.\"}"
```
Comment (mkuron) [inline maintainer/gh_post_status.sh]: Depending on whether `$STATUS` is `pending`, `success` or `failure`, the description should be different.

**Review by mkuron — CHANGES_REQUESTED** [review]

File: src/core/communication.cpp [hunk 2]
```diff
@@ -1864,7 +1864,7 @@ void mpi_send_exclusion_slave(int part1, int part2) {
 }
 
 /************** REQ_SET_FLUID **************/
-void mpi_send_fluid(int node, int index, double rho, double *j, double *pi) {
+void mpi_send_fluid(int node, int index, double rho, const std::array<double, 3> j, const std::array<double, 6> pi) {
```
Comment (fweik) [inline src/core/communication.cpp]: Is there a specific reason to take copies of j and pi here?

Comment (KaiSzuttor) [reply src/core/communication.cpp]: no, fixed

File: maintainer/gh_post_status.sh (hunk 1 above)
Comment (KaiSzuttor) [reply maintainer/gh_post_status.sh]: was a test, removed it

**Comment by fweik** [comment]
Can you please check if the CPU test works with #2136?

**Comment by KaiSzuttor** [comment]
works...

**Comment by KaiSzuttor** [comment]
@fweik i think it's ready for review

**Comment by fweik** [comment]
Did you look at the latest changes in #2136 ? I think that one has to go first.

**Pull request merged**